Runtime/AvatarCreator/GenderSelectElement.cs

```diff
@@ -0,0 +1,45 @@
+using ReadyPlayerMe.Core;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public class GenderSelectElement : MonoBehaviour
+    {
+        [Header("Buttons")]
+        [SerializeField, Tooltip("Button for selecting a male avatar")]
+        private Button maleButton;
+
+        [SerializeField, Tooltip("Button for selecting a female avatar")]
+        private Button femaleButton;
+
+        [Space(5)]
+        [Header("Events")]
+        public UnityEvent<OutfitGender> OnGenderSelected;
+
+        private void Awake()
+        {
+            InitializeButtons();
+        }
+
+        private void InitializeButtons()
+        {
+            if (maleButton != null)
+            {
+                maleButton.onClick.AddListener(() => OnGenderSelected?.Invoke(OutfitGender.Masculine));
+            }
+
+            if (femaleButton != null)
+            {
+                femaleButton.onClick.AddListener(() => OnGenderSelected?.Invoke(OutfitGender.Feminine));
+            }
+        }
+
+        private void OnDestroy()
+        {
+            maleButton?.onClick.RemoveListener(() => OnGenderSelected?.Invoke(OutfitGender.Masculine));
```

Comment:
Does this actually work to remove the listener? As they're actually both separate anonymous methods, so technically you're removing nothing? Unless Unity has some magic to match them up?

Would it work better to not have these as anonymous methods, and then add and remove the listener by reference to the method? Like suggested here: https://forum.unity.com/threads/how-do-i-remove-button-listeners-and-should-i.1318596/

```suggestion
            maleButton?.onClick.RemoveListener(SendMaleSelected);
            
            private void SendMaleSelected() {
               OnGenderSelected?.Invoke(OutfitGender.Masculine)
            }
```

Reply:
It's a good point actually and I think you're right, it was a bit of copy pasta from chatgpt :D 
I will update to use a function just to be sure.

Reply:
updated